### ISSUE TITLE
Loss formulation of Strategy-B: Alignment using red data

### ISSUE THREAD

**Issue by yunjae-won** [declare-lab/red-instruct#4]
Hi! Thank you for the amazing paper! I have a question regarding your paper (Red-Teaming Large Language Models using Chain of Utterances for Safety-Alignment). Regarding the loss (Equation 3 from the paper) for **Strategy-B: Alignment using red data**, it is mentioned that we partition the red samples based on the negative log-likelihood(NLL). However, the paper seems to compute the NLL of a sample by summing the token level log probabilities (sum of log-probabilities is equivalent to the log of product of probabilities). Then are we partitioning the red samples based on the sum of NLL? In other words, are we partitioning the red samples based on whether the sum of token level log probabilities exceed 1? This seems unnatural given that the sum of NLL for some sample is typically larger than 50. I think the threshold value of 1.0 is intended to be compared to the token-level NLL, thus I think the correct formulation should be that we partition the red samples based on the _average_ (not sum) of token-level NLL.

**Comment by Bhardwaj-Rishabh** [comment]
Hi @YJWon99, thanks for pointing it out! Yes, you are right, we partitioned samples based on average token-level NLL. To compute this per sample, we used the cross-entropy loss function from torch.nn with reduction attribute set to 'mean'. We will make corrections in the updated version of the paper. Thanks again!
